libs/api/domains/national-registry-v3/src/lib/nationalRegistryV3.service.ts

```diff
@@ -0,0 +1,310 @@
+import {
+  EinstaklingurDTOAllt,
+  NationalRegistryV3ClientService,
+} from '@island.is/clients/national-registry-v3'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import type { Logger } from '@island.is/logging'
+import { Person } from './graphql/models/nationalRegistryPerson.model'
+import { User } from '@island.is/auth-nest-tools'
+import { ChildGuardianship } from './graphql/models/nationalRegistryChildGuardianship.model'
+import { Spouse } from './graphql/models/nationalRegistrySpouse.model'
+import { Inject } from '@nestjs/common'
+import { Address } from './graphql/models/nationalRegistryAddress.model'
+import { Birthplace } from './graphql/models/nationalRegistryBirthplace.model'
+import { Citizenship } from './graphql/models/nationalRegistryCitizenship.model'
+import { Name } from './graphql/models/nationalRegistryName.model'
+import { Religion } from './graphql/models/nationalRegistryReligion.model'
+import { Custodian } from './graphql/models/nationalRegistryCustodian.model'
+import { DomicilePopulace } from './graphql/models/nationalRegistryDomicilePopulace.model'
+
+type ExcludesFalse = <T>(x: T | null | undefined | '') => x is T
+
+export class NationalRegistryV3Service {
+  constructor(
+    private nationalRegistryApi: NationalRegistryV3ClientService,
+    @Inject(LOGGER_PROVIDER)
+    private logger: Logger,
+  ) {}
+
+  private fetchData = (nationalId: string) =>
+    this.nationalRegistryApi.getData(nationalId)
+
+  private extractPerson = (data: EinstaklingurDTOAllt) => {
+    if (!data.kennitala || !data.nafn) {
+      return null
+    }
+
+    return (
+      data && {
+        nationalId: data.kennitala,
+        fullName: data.nafn,
+        gender: data.kyn?.kynTexti,
+        familyRegistrationCode: data.logheimilistengsl?.logheimilistengsl,
+        banMarking: data.bannmerking === 'true',
+        fate: data.afdrif,
+      }
+    )
+  }
+
+  async getNationalRegistryPerson(nationalId: string): Promise<Person | null> {
+    const person = await this.fetchData(nationalId)
+
+    return this.extractPerson(person)
+  }
+
+  async getAddress(nationalId: string): Promise<Address | null> {
+    const data = await this.fetchData(nationalId)
```

Comment:
Are all of the field resolvers fetching the same person again and again?

You could include the raw data in a private optional field of the `Person` type (eg `rawData`) so you don't have to fetch the data again in the field resolvers.

Reply:
yeah this was a bit undercooked, i'll fix